File: src/java_dev/views/PanelTweets.java
package java_dev.views;

import java.awt.*;
import java.net.MalformedURLException;
import java.net.URL;

import javax.swing.GroupLayout;
import javax.swing.ImageIcon;
import javax.swing.JLabel;
import javax.swing.JList;
import javax.swing.JPanel;
import javax.swing.JTextField;
import javax.swing.LayoutStyle;
import javax.swing.ListCellRenderer;
import twitter4j.Status;

public class PanelTweets extends JPanel implements ListCellRenderer<Status> {
  private JLabel jLabel1;
  private JLabel jLabel2;
  private JLabel jLabel3;
  private JTextField jTextField1;

  public PanelTweets() {
    initComponents();
  }

  private void initComponents() {
    jLabel1 = new JLabel();
    jLabel2 = new JLabel();
    jTextField1 = new JTextField();
    jLabel3 = new JLabel();

    jLabel1.setText("jLabel1");
    jLabel2.setText("jLabel2");
    jLabel2.setMaximumSize(new java.awt.Dimension(48, 48));
    jLabel2.setMinimumSize(new java.awt.Dimension(48, 48));
    jLabel2.setPreferredSize(new java.awt.Dimension(48, 48));
    jTextField1.setText("jTextField1");
    jLabel3.setFont(new java.awt.Font("Ubuntu", Font.ITALIC, 12));
    jLabel3.setText("jLabel3");

    GroupLayout layout = new GroupLayout(this);

    this.setLayout(layout);
    layout.setHorizontalGroup(layout.createParallelGroup(GroupLayout.Alignment.LEADING).addGroup(layout
        .createSequentialGroup().addContainerGap()
        .addGroup(layout.createParallelGroup(GroupLayout.Alignment.LEADING)
            .addGroup(layout.createSequentialGroup().addComponent(jLabel1).addGap(0, 0, Short.MAX_VALUE))
            .addGroup(layout.createSequentialGroup()
                .addComponent(jLabel2, GroupLayout.PREFERRED_SIZE, GroupLayout.DEFAULT_SIZE, GroupLayout.PREFERRED_SIZE)
                .addGap(18, 18, 18)
                .addGroup(layout.createParallelGroup(GroupLayout.Alignment.LEADING)
                    .addGroup(layout.createSequentialGroup().addComponent(jLabel3).addGap(0, 0, Short.MAX_VALUE))
                    .addComponent(jTextField1, GroupLayout.DEFAULT_SIZE, 310, Short.MAX_VALUE))))
        .addContainerGap()));
    layout
        .setVerticalGroup(layout.createParallelGroup(GroupLayout.Alignment.LEADING)
            .addGroup(layout.createSequentialGroup().addContainerGap().addComponent(jLabel1).addGap(18, 18, 18)
                .addGroup(layout.createParallelGroup(GroupLayout.Alignment.BASELINE)
                    .addComponent(jLabel2, GroupLayout.PREFERRED_SIZE, GroupLayout.DEFAULT_SIZE,
                        GroupLayout.PREFERRED_SIZE)
                    .addComponent(jTextField1, GroupLayout.PREFERRED_SIZE, GroupLayout.DEFAULT_SIZE,
                        GroupLayout.PREFERRED_SIZE))
                .addPreferredGap(LayoutStyle.ComponentPlacement.RELATED, 9, Short.MAX_VALUE).addComponent(jLabel3)
                .addContainerGap()));
  }

  @Override
  public Component getListCellRendererComponent(JList<? extends Status> list, Status value, int index,
      boolean isSelected, boolean cellHasFocus) {
    jLabel1.setText("@" + value.getUser().getScreenName());
    jLabel3.setText(value.getCreatedAt() + "");
    jTextField1.setText(value.getText());

    URL urlIMG;

    try {
      urlIMG = new URL(value.getUser().getProfileImageURL());

      ImageIcon img = new ImageIcon(urlIMG);
      jLabel2.setIcon(img);
    } catch (MalformedURLException e) {
      e.printStackTrace();
    }

    return this;
  }
}

File: src/kotlin_dev/views/PanelTweets.kt
package kotlin_dev.views

import twitter4j.Status
import java.awt.Dimension
import java.awt.Font
import java.net.URL
import javax.swing.*

class PanelTweets : ListCellRenderer<Status>, JPanel() {
    private var jLabel1: JLabel = JLabel("jLabel1")
    private var jLabel2: JLabel = JLabel("jLabel2")
    private var jLabel3: JLabel = JLabel("jLabel3")
    private var jTextField1: JTextField = JTextField("jTextField1")

    init {
        initComponents()
    }

    private fun initComponents() {
        jLabel2.maximumSize = Dimension(48, 48)
        jLabel2.minimumSize = Dimension(48, 48)
        jLabel2.preferredSize = Dimension(48, 48)
        jLabel3.font = Font("Ubuntu", 2, 12)

        val layout = GroupLayout(this)
        this.layout = layout

        layout.setHorizontalGroup(
            layout.createParallelGroup(GroupLayout.Alignment.LEADING).addGroup(
                layout
                    .createSequentialGroup().addContainerGap()
                    .addGroup(
                        layout.createParallelGroup(GroupLayout.Alignment.LEADING)
                            .addGroup(
                                layout.createSequentialGroup().addComponent(jLabel1)
                                    .addGap(0, 0, Short.MAX_VALUE.toInt())
                            )
                            .addGroup(
                                layout.createSequentialGroup()
                                    .addComponent(
                                        jLabel2,
                                        GroupLayout.PREFERRED_SIZE,
                                        GroupLayout.DEFAULT_SIZE,
                                        GroupLayout.PREFERRED_SIZE
                                    )
                                    .addGap(18, 18, 18)
                                    .addGroup(
                                        layout.createParallelGroup(GroupLayout.Alignment.LEADING)
                                            .addGroup(
                                                layout.createSequentialGroup().addComponent(jLabel3)
                                                    .addGap(0, 0, Short.MAX_VALUE.toInt())
                                            )
                                            .addComponent(
                                                jTextField1,
                                                GroupLayout.DEFAULT_SIZE,
                                                310,
                                                Short.MAX_VALUE.toInt()
                                            )
                                    )
                            )
                    )
                    .addContainerGap()
            )
        )
        layout
            .setVerticalGroup(
                layout.createParallelGroup(GroupLayout.Alignment.LEADING)
                    .addGroup(
                        layout.createSequentialGroup().addContainerGap().addComponent(jLabel1).addGap(18, 18, 18)
                            .addGroup(
                                layout.createParallelGroup(GroupLayout.Alignment.BASELINE)
                                    .addComponent(
                                        jLabel2, GroupLayout.PREFERRED_SIZE, GroupLayout.DEFAULT_SIZE,
                                        GroupLayout.PREFERRED_SIZE
                                    )
                                    .addComponent(
                                        jTextField1, GroupLayout.PREFERRED_SIZE, GroupLayout.DEFAULT_SIZE,
                                        GroupLayout.PREFERRED_SIZE
                                    )
                            )
                            .addPreferredGap(LayoutStyle.ComponentPlacement.RELATED, 9, Short.MAX_VALUE.toInt())
                            .addComponent(jLabel3)
                            .addContainerGap()
                    )
            )
    }

    override fun getListCellRendererComponent(
        list: JList<out Status?>?, value: Status, index: Int,
        isSelected: Boolean, cellHasFocus: Boolean
    ): PanelTweets {
        val status: Status = value

        jLabel1.text = "@ ${status.user.screenName}"
        jLabel3.text = status.createdAt.toString()
        jTextField1.text = status.text

        val urlIMG = URL(status.user.profileImageURL)
        val img = ImageIcon(urlIMG)

        jLabel2.icon = img

        return this
    }
}